File: backend/src/health/health.controller.ts
import { Controller, Get } from '@nestjs/common';
import { ApiTags, ApiOperation, ApiResponse } from '@nestjs/swagger';

@ApiTags('health')
@Controller('health')
export class HealthController {
  @Get()
  @ApiOperation({ summary: 'Health check endpoint' })
  @ApiResponse({ status: 200, description: 'API is running' })
  check() {
    return {
      status: 'ok',
      message: 'FuelMate API is running',
      timestamp: new Date().toISOString(),
    };
  }
}

File: backend-dotnet/Controllers/HealthController.cs
using Microsoft.AspNetCore.Mvc;

namespace FuelMateBackend.Controllers;

[ApiController]
[Route("[controller]")]
[Tags("health")]
public class HealthController : ControllerBase
{
    /// <summary>
    /// Health check endpoint
    /// </summary>
    [HttpGet]
    public IActionResult Check()
    {
        return Ok(new
        {
            status = "ok",
            message = "FuelMate API is running",
            timestamp = DateTime.UtcNow.ToString("O")
        });
    }
}

